pkg/ds/farm.go

```diff
@@ -564,6 +565,25 @@ func (dsf *Farm) spawnDaemonSet(
 
 	desiresCh := ds.WatchDesires(quitSpawnCh, updatedCh, deletedCh)
 
+	go func() {
+		aggregateHealth := replication.AggregateHealth(ds.PodID(), *dsf.healthChecker)
```

Comment:
Consider this possibility - we have this `AggregateHealth` call that is used to *always* monitor the health (to emit the metric).

When we go enact a replication, we make **another** `AggregateHealth` call that creates a separate watch.

**Why not just pass this aggregator to the replication**?

It is entirely possible. ds is the only package using replication at the moment, so I feel we are relatively free to change the interface.